cfme/ansible_tower/jobs.py

```diff
@@ -51,15 +56,82 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(TowerJobsView):
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        return self.title.text == '{} (Summary)'.format(self.context['object'].template_name)
+
+
 @attr.s
 class TowerJobs(BaseEntity):
-    pass
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of("Status")
+
+    def delete(self):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Job', handle_alert=True)
+
+    @property
+    def is_job_finished(self):
+        return True if self.status == "successful" else False
 
 
 @attr.s
 class TowerJobsCollection(BaseCollection):
     ENTITY = TowerJobs
 
+    def all(self):
+        view = navigate_to(self, 'All')
+        jobs = []
+        template = self.filters.get('temp_name')
+        for entity in [entity for entity in view.entities.get_all(surf_pages=True)]:
+            jobs.append(self.instantiate(template_name=template))
+
+        return jobs
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'All')
+        template = self.filters.get('temp_name')
+        for row in view.entities.elements:
+            if row.template_name.text == template:
+                return row.status.text
+        return ''
+
+    @property
+    def is_all_finished(self):
```

Comment:
Grammar nazi incoming, 
```suggestion
    def are_all_finished(self):
```